geojson_modelica_translator/system_parameters/system_parameters.py

```diff
@@ -112,14 +111,14 @@ def resolve_paths(self):
         """Resolve the paths in the file to be absolute if they were defined as relative. This method uses
         the JSONPath (with ext) to find if the value exists in the self.data object. If so, it then uses
         the set_param which navigates the JSON tree to set the value as needed."""
-        filepath = os.path.abspath(os.path.dirname(self.filename))
+        filepath = Path(self.filename).parent.resolve()
 
         for pe in self.PATH_ELEMENTS:
             matches = parse(pe["json_path"]).find(self.data)
             for index, match in enumerate(matches):
                 # print(f"Index {index} to update match {match.path} | {match.value} | {match.context}")
-                new_path = os.path.join(filepath, match.value)
-                parse(str(match.full_path)).update(self.data, new_path)
+                new_path = Path(filepath) / match.value
+                parse(str(match.full_path)).update(self.data, str(new_path.as_posix()))
```

Comment:
`as_posix` returns a string.

```suggestion
                parse(str(match.full_path)).update(self.data, new_path.as_posix())
```

Reply:
So it does. My earlier test config must have been different in some other way. Thanks Mostapha.